### PR TITLE
Restores pre-root boiler dig mechanic

### PR DESCRIPTION
**Pull request by Barnet2**
## About The Pull Request
Basically when you select bombard you take 3 seconds to dig in, but can move freely at any time during this. This was behavior removed with root that I forgot to bring back.
## Why It's Good For The Game
Should have been brought back with root removal to restore pre-root behavior.
## Changelog
:cl:
balance: Restores pre-root boiler dig mechanic, have to sit still and wait 3 seconds before firing a glob (you can still move, but cancels the channel)
/:cl:

### CODE REVIEW

**Comment by MiniMeatwad** [comment]
Oh lol

**Comment by Xander3359** [comment]
Wait this whole time bombard had no channel? So boiler was giga buffed? LMAOOOOOOOO

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm [hunk 1]
```diff
@@ -202,6 +227,24 @@ GLOBAL_LIST_INIT(boiler_glob_image_list, list(
...
+/mob/living/carbon/xenomorph/boiler/Moved(atom/OldLoc,Dir)
+	. = ..()
+	if(selected_ability?.type == /datum/action/ability/activable/xeno/bombard)
+		var/datum/action/ability/activable/bomb = actions_by_path[/datum/action/ability/activable/xeno/bombard]
+		bomb.on_deselection()
+		selected_ability.button.icon_state = "template"
+		selected_ability = null
+		update_action_button_icons()
+
+/mob/living/carbon/xenomorph/boiler/proc/set_bombard_pointer()
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm:240]: autodoc

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm [hunk 2]
```diff
@@ -202,6 +227,24 @@ GLOBAL_LIST_INIT(boiler_glob_image_list, list(
...
+	if(client)
+		client.mouse_pointer_icon = 'icons/mecha/mecha_mouse.dmi'
+
+/mob/living/carbon/xenomorph/boiler/proc/reset_bombard_pointer()
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm:244]: autodoc

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm [hunk 3]
```diff
@@ -188,12 +188,37 @@ GLOBAL_LIST_INIT(boiler_glob_image_list, list(
...
+	X.visible_message(span_notice("\The [X] digs itself into the ground!"), \
+		span_notice("We dig ourselves into place! If we move, we must wait again to fire."), null, 5)
+	X.set_bombard_pointer()
 	RegisterSignal(owner, COMSIG_MOB_ATTACK_RANGED, TYPE_PROC_REF(/datum/action/ability/activable/xeno/bombard, on_ranged_attack))
 
 /datum/action/ability/activable/xeno/bombard/on_deselection()
+	var/mob/living/carbon/xenomorph/boiler/X = owner
+	if(X.selected_ability == src)
+		X.reset_bombard_pointer()
+		to_chat(X, span_notice("We relax our stance."))
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm:221]: don't use X

**Review by Lumipharon — CHANGES_REQUESTED** [review]

**Comment by Helg2** [comment]
> I believe that a caste like boiler, which can completely destroy marine cohesion, destroy cades and deal lethal damage OFFSCREEN should have a lot more risk.

If only the game had something like a marine high damage offscreen weapon, which is totally free, and with which marines totally cannot cooperate to kill xenos in a few seconds. HMMMMMMMMMMMMMM.

**Comment by mkvrs** [comment]
> If only the game had something like a marine high-damaging offscreen weapon, which is totally free, and with which marines totally cannot cooperate to kill xenos in a few seconds. HMMMMMMMMMMMMMM.

Hmm yes...!
sr 127?
Sure it'll make them go away for a while, shooting at their general direction does that too.
But does it stop them from just ignoring the maybe one or two sniper rifle users that DO bother to target boilers? 
No it doesn't, they just peek out, bombard and repeat.
''and with which marines totally cannot cooperate to kill xenos in a few seconds.''
Please tell me what you would do, flank it? Go behind it?  These maps don't allow such things, xenos don't allow such things to happen. The only option left would be to bumrush the boiler which either results in marines getting obliterated by the hive or the boiler just running away in mach 5.

**Comment by Helg2** [comment]
> Please tell me what you would do

3 cooperated overcharged shots from the laser sniper rifle, will most probably obliterate the boiler. 
If you can't simply cooperate with others, this is simply a skill issue.
If you either can't flank the boiler, this is again a skill issue.

**Comment by coldironwarrior** [comment]
if we do this, would it be possible (although i know it's not original behavior) to remove the wind-up from zoom?

root + zoom was 4 seconds to initiate; this + zoom is 4 seconds to initiate, and i feel like wind-up on zoom is goofy and all wind-up should be on bombard ability (as is here). i'd be down for the whole One Second less of prep time required too, adding dig back is pretty significant for counter-balancing already, i feel.

**Comment by mkvrs** [comment]
> 3 cooperated overcharged shots from the laser sniper rifle, will most probably obliterate the boiler. If you can't simply cooperate with others, this is simply a skill issue. If you either can't flank the boiler, this is again a skill issue.

3 Laser users that somehow, wont FF and hit shots on the boiler at the same time without the boiler running away? They don't have root anymore, how will this work?
You will never see this, you won't see it happen with AMR's either.
And on the matter of flanking, please think about how they can see you before you see them.
The only viable and easily applicable solution to boilers ends up being artillery, which often gets cucked by the maps and co-operation
Though when it isn't cucked it kind of does the job.

**Review by Lumipharon — CHANGES_REQUESTED** [review]
Bruh, I said move it to the bottom, there are still other procs for the ability under where you've put it lol.

ALL existing proc definitions, then the new proc, then the related new boiler procs

**Review by Kuro020 — APPROVED** [review]

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**